src/main/java/org/jenkinsci/plugins/KeycloakAccess.java

```diff
@@ -122,7 +132,7 @@ public UserDetails authenticate( String username, String password ) throws Authe
 		HttpResponse usersResponse = keycloakDeployment.getClient().execute( getUsersReq );
 		int statusCode = usersResponse.getStatusLine().getStatusCode();
 		if ( statusCode != 200 ) {

```

Comment:
Can we also have it cache when a user is not found in Keycloak? Normally if a user isn't found once that will (probably) never be found -- but at least caching it with a timeout still would prevent some additional hits to KC.

Reply:
I dont think I will cache here in authenticate as they could have failed authentication.